lib/ansible/modules/cloud/amazon/aws_api_gateway.py

```diff
@@ -0,0 +1,241 @@
+     the same using a tag embedded in the API name.
+
+'''
+
+EXAMPLES = '''
+# Update API resources for development
+tasks:
+- name: update API
+  aws_api_gateway:
+    api_id: 'abc123321cba'
+    state: present
+    swagger_file: my_api.yml
+
+# update definitions and deploy API to production
+tasks:
+- name: deploy API
+  aws_api_gateway:
+    api_id: 'abc123321cba'
+    state: present
+    swagger_file: my_api.yml
+    stage: production
+    deploy_desc: Make auth fix available.
+'''
+
+RETURN = '''
+output:
+  description: the data returned by put_restapi in boto3
+  returned: success
+  type: dict
+  sample:
+    'data':
+      {
+          "id": "abc123321cba",
+          "name": "MY REST API",
+          "createdDate": 1484233401
+      }
+'''
+
+import json
+from ansible.module_utils.basic import AnsibleModule, traceback
+from ansible.module_utils.ec2 import ec2_argument_spec, get_aws_connection_info, boto3_conn, camel_dict_to_snake_dict
+
+from ansible.module_utils.ec2 import HAS_BOTO3
+
+try:
+    import botocore
+    HAS_BOTOCORE = True
+except ImportError:
+    HAS_BOTOCORE = False
+
+
+def main():
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(
+        dict(
+            api_id=dict(type='str', required=False),
+            state=dict(type='str', default='present', choices=['present', 'absent']),
+            swagger_file=dict(type='path', default=None, aliases=['src', 'api_file']),
+            swagger_dict=dict(type='json', default=None),
+            swagger_text=dict(type='str', default=None),
+            stage=dict(type='str', default=None),
+            deploy_desc=dict(type='str', default=None),
+        )
+    )
+
+    mutually_exclusive = [['swagger_file', 'swagger_dict', 'swagger_text' ]]  # noqa: F841
+
+    module = AnsibleModule(argument_spec=argument_spec,
+                           supports_check_mode=False,  )   # TODO !!!`
+
+    api_id = module.params.get('api_id')
+    state = module.params.get('state')   # noqa: F841
+    swagger_file = module.params.get('swagger_file')
+    swagger_dict = module.params.get('swagger_dict')
+    swagger_text = module.params.get('swagger_text')
+    stage= module.params.get('stage')
+    deploy_desc= module.params.get('deploy_desc')
+
+#    check_mode = module.check_mode
+    changed = False
+
+    if not HAS_BOTO3:
+        module.fail_json(msg='Python module "boto3" is missing, please install boto3')
+
+    if not HAS_BOTOCORE:
+        module.fail_json(msg='Python module "botocore" is missing, please install it')
+
+    region, ec2_url, aws_connect_kwargs = get_aws_connection_info(module, boto3=True)
+    if not region:
+        module.fail_json(msg='region must be specified')
+
+    try:
+        client = boto3_conn(module, conn_type='client', resource='apigateway',
+                            region=region, endpoint=ec2_url, **aws_connect_kwargs)
+    except (botocore.exceptions.ValidationError) as e:
+        msg="Incorrect AWS configuration"
```

Comment:
There are a lot on non PEP8 assignations here.

Basically, in a dictionary or method call, there shouldn't be a space around the `=` operator. For assignments, there should be a space around the `=` call. So line 171 here is wrong, lines 168 and 172 are right